cpp/src/fil/fil.cu

```diff
@@ -134,7 +134,21 @@ struct forest {
     max_shm_ = std::min(max_shm_, max_shm_std);
   }
 
-  void init_common(const forest_params_t* params) {
+  void init_fixed_block_count(const raft::handle_t& h, int blocks_per_sm) {
```

Comment:
It would be better to fuse it together with `init_max_shm()` (feel free to do it in a different pull request).

Reply:
yes, let's do this in https://github.com/rapidsai/cuml/pull/3056